runtime/syntax/generator/vim.vim.base

```diff
@@ -862,218 +863,266 @@ syn region	vimHiLink	contained matchgroup=Type start="\%(\<hi\%[ghlight]!\=\s\+\
 else
- syn region vimEmbedError start=+py\%[thon]3\=\s*<<\s*\z(.*\)$+ end=+^\z1$+
- syn region vimEmbedError start=+py\%[thon]3\=\s*<<\s*$+ end=+\.$+
+ syn region vimEmbedError start=+\<py[3x]\=f\%[ile]\>+ end=+$+
+ syn region vimEmbedError start=+\<\%(py\%[thon]\|py\%(thon\)\=[3x]\|py[3x]\=do\)\>+ skip=+\n\s*\\\|\n\s*["#]\\ + end=+$+
+ syn region vimEmbedError start=+\<\%(py\%[thon]\|py\%(thon\)\=[3x]\)\=\s*<<\s*\%(trim\s\+\)\=\z(\S\+\)\s*$+      end=+^\s*\z1$+
+ syn region vimEmbedError start=+\<\%(py\%[thon]\|py\%(thon\)\=[3x]\)\=\s*<<\s*\%(trim\s*\)\=$+	      end=+^\s*\.$+
 endif
 unlet s:pythonpath
 
+" [-- ruby --] {{{3
+let s:rubypath = fnameescape(expand("<sfile>:p:h") .. "/ruby.vim")
+if !filereadable(s:rubypath)
+ for s:rubypath in split(globpath(&rtp, "syntax/ruby.vim"), "\n")
+  if filereadable(fnameescape(s:rubypath))
+   let s:rubypath = fnameescape(s:rubypath)
+   break
+  endif
+ endfor
+endif
+if g:vimsyn_embed =~# 'r' && filereadable(s:rubypath)
+ unlet! b:current_syntax
+ exe "syn include @vimRubyScript " .. s:rubypath
+ syn keyword vimRuby rub[y]   skipwhite nextgroup=vimRubyRegion,vimRubyStatement
+ syn keyword vimRuby rubyd[o] skipwhite nextgroup=vimRubyStatement
+ syn keyword vimRuby rubyf[ile]
+ syn region  vimRubyStatement contained start="\S" skip=+\n\s*\\\|\n\s*["#]\\ + end="$" contains=@vimRubyScript,@vimContinue
+ VimFoldr syn region vimRubyRegion	contained matchgroup=vimScriptHereDocStart start=+<<\s*\z(\S\+\)\ze\s*$+         matchgroup=vimScriptHereDocStop end=+^\z1$+	  contains=@vimRubyScript
+ VimFoldr syn region vimRubyRegion	contained matchgroup=vimScriptHereDocStart start=+<<\ze\s*$+	         matchgroup=vimScriptHereDocStop end=+^\.$+	  contains=@vimRubyScript
+ VimFoldr syn region vimRubyRegion	contained matchgroup=vimScriptHereDocStart start=+<<\s*trim\s\+\z(\S\+\)\ze\s*$+ matchgroup=vimScriptHereDocStop end=+^\s*\z1$+ contains=@vimRubyScript
+ VimFoldr syn region vimRubyRegion	contained matchgroup=vimScriptHereDocStart start=+<<\s*trim\ze\s*$+	         matchgroup=vimScriptHereDocStop end=+^\s*\.$+  contains=@vimRubyScript
+else
+ syn region vimEmbedError start=+\<rubyf\%[ile]\>+ end=+$+
+ syn region vimEmbedError start=+\<\%(ruby\=\|rubydo\=\)\>+ skip=+\n\s*\\\|\n\s*["#]\\ + end=+$+
+ syn region vimEmbedError start=+\<ruby\=\s*<<\s*\%(trim\s\+\)\=\z(\S\+\)\s*$+ end=+^\s*\z1$+
+ syn region vimEmbedError start=+\<ruby\=\s*<<\s*\%(trim\s*\)\=$+	       end=+^\s*\.$+
+endif
+unlet s:rubypath
+
 " [-- tcl --] {{{3
 if has("win32") || has("win95") || has("win64") || has("win16")
  " apparently has("tcl") has been hanging vim on some windows systems with cygwin
- let s:trytcl= (&shell !~ '\<\%(bash\>\|4[nN][tT]\|\<zsh\)\>\%(\.exe\)\=$')
+ let s:trytcl = (&shell !~ '\<\%(bash\>\|4[nN][tT]\|\<zsh\)\>\%(\.exe\)\=$')
 else
- let s:trytcl= 1
+ let s:trytcl = 1
 endif
```

Comment:
I'm wondering if we still need this part.
Trying to use Cygwin version of Tcl from Win32 version of Vim might cause hanging long time ago. But nowadays, Cygwin DLLs have the cyg prefix, so I don't think it can happen.